src/components/ProductFilter.js

```diff
@@ -0,0 +1,278 @@
+        <h4 className="productFilter_filterHeading">Filter By Rating</h4>
+        <div className="productListing_options">
+          {[1, 2, 3, 4].map((entry, index) => {
+            return (
+              <label key={index}>
+                {entry}+ Star
+                <input
+                  type="checkbox"
+                  onChange={() => dispatch({ type: "rating", payload: entry })}
+                  checked={state.rating === entry}
+                />
+              </label>
+            );
+          })}{" "}
+          <label>
+            None
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "rating", payload: 0 })}
+              checked={state.brand === 0}
+            />
+          </label>
+        </div>
+      </div>
+
+      <div className="productFilter_filter">
+        <h4 className="productFilter_filterHeading">Filter Further</h4>
+        <div className="productListing_options">
+          <label>
+            Only Fast Delivery
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "deliveryToggle" })}
+              checked={state.onlyFastDelivery}
+            />
+          </label>
+          <label>
+            In Stock
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "stockToggle" })}
+              checked={state.onlyInStock}
+            />
+          </label>
+          <label>
+            None
+            <input
+              type="checkbox"
+              onChange={() => dispatch({ type: "resetstockndelivery" })}
+              checked={!state.onlyFastDelivery && !state.onlyInStock}
+            />
+          </label>
+        </div>
+      </div>
+
+      {/* helo */}
+    </div>
+  );
+};
+
+const ProductFilterResponsive = (Component) => {
+  /*
+  this is what we call a higher order component (HOC) basically its wrapping
+  the componenet which im passign it into a another component which will extend its functionality
+  */
+
+  //please ignore this as this is not finished yet just think of it as a function which is going to return the component
+  // as it is which is being passed down to it
+  const OutputComponent = (props) => {
+    const [mobile, setMobile] = useState(false);
+    const [show, setShow] = useState(false);
+
+    useEffect(() => {
+      if (window.screen.width <= 768) setMobile(true);
+      else setMobile(false);
+    }, [window.screen.width]);
+
+    return mobile ? (
+      <>
+        <div className="Filterresponsive">
+          {show && (
+            <Component {...props} className="productFilter--responsive" />
+          )}
+          <button className="btn" onClick={() => setShow((prev) => !prev)}>
+            Filter
+          </button>
+        </div>
+      </>
+    ) : (
+      <Component {...props} />
+    );
+  };
+
+  return OutputComponent;
+};
+
+export const useCustomFilter = (data, state) => {
```

Comment:
This hook can also be extracted out to a separate file. Ensure that your files does not have more than 100 lines. keep a hard stop at 150 .. if it goes more than probably you can extract out the something to a separate file

Reply:
great will keep that in mind